core/dbt/task/compile.py

```diff
@@ -61,23 +61,27 @@ def get_runner_type(self, _):
         return CompileRunner
 
     def task_end_messages(self, results):
-        if getattr(self.args, "inline", None):
-            result = results[0]
-            fire_event(
-                CompiledNode(node_name=result.node.name, compiled=result.node.compiled_code)
-            )
+        is_inline = bool(getattr(self.args, "inline", None))
 
-        if self.selection_arg:
+        if is_inline:
+            matched_results = [result for result in results if result.node.name == "inline_query"]
+        elif self.selection_arg:
             matched_results = [
-                result for result in results if result.node.name == self.selection_arg[0]
+                result for result in results if result.node.name in self.selection_arg
             ]
-            if len(matched_results) == 1:
-                result = matched_results[0]
-                fire_event(
-                    CompiledNode(node_name=result.node.name, compiled=result.node.compiled_code)
-                )
+        # No selector passed, compiling all nodes
+        else:
+            matched_results = []
 
-        fire_event(CompileComplete())
+        for result in matched_results:
```

Comment:
@jtcohen6 did we agree on firing one event for each selected node is a good idea? what if we accidentally selected 1000 node?

Reply:
If I had my druthers, I'd prefer we do this in a way that kept more of this logic within dbt's selection syntax, e.g. "we only want to support the [`FQN` selector method](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/graph/selector_methods.py#L34)."

As it is, we're going to apply the selection syntax, and then filter it down to only preview the nodes whose names explicitly appear in the `--select` arg. This means the `show` command won't support yaml selectors, `tag:`-based selection (even if only one node has that tag), etc. We also won't be showing a log message explaining why, if a node is selected, it's not being previewed.
```
$ dbt show -s tag:doesnt_exist
17:26:05  Running with dbt=1.5.0-b4
17:26:05  Found 1 model, 0 tests, 0 snapshots, 1 analysis, 420 macros, 0 operations, 1 seed file, 0 sources, 0 exposures, 0 metrics, 0 groups
17:26:05  The selection criterion 'tag:doesnt_exist' does not match any nodes
17:26:05
17:26:05  Nothing to do. Try checking your model configs and model specification args
$ dbt show -s tag:one_node_has_this_tag
17:26:20  Running with dbt=1.5.0-b4
17:26:21  Found 1 model, 0 tests, 0 snapshots, 1 analysis, 420 macros, 0 operations, 1 seed file, 0 sources, 0 exposures, 0 metrics, 0 groups
17:26:21
17:26:21  Concurrency: 5 threads (target='dev')
17:26:21
```
In terms of the "happy path" for the `show` command, the intent is to only `show` resource(s) explicitly specified in the `--select` syntax. Could I ask that we at least fire an event here, if a set of nodes is returned from the selection syntax, and then filtered out because the node's name wasn't explicitly passed to `--select`?
```
$ dbt show -s tag:one_node_has_this_tag
17:26:20  Running with dbt=1.5.0-b4
17:26:21  Found 1 model, 0 tests, 0 snapshots, 1 analysis, 420 macros, 0 operations, 1 seed file, 0 sources, 0 exposures, 0 metrics, 0 groups
17:26:21  The selection criterion 'tag:one_node_has_this_tag' selected one node, but the 'show' command will only preview models explicitly named in the 'select' argument
17:26:21
17:26:21  Concurrency: 5 threads (target='dev')
17:26:21
```
If we can do that, then this is fine by me.

Reply:
Is this comment addressed? @aranke